api-authorization/server/authorization-middleware.js

```diff
@@ -0,0 +1,38 @@
+const jwt = require('jsonwebtoken'); // eslint-disable-line
+const ClientError = require('./client-error'); // eslint-disable-line
+
+function authorizationMiddleware(req, res, next) {
+
+  const xToken = req.get('X-Access-Token');
+  // console.log('token', xToken);
+  if (!xToken) {
+    throw new ClientError(401, 'authentication required');
+  }
+
+  try {
+    const decoded = jwt.verify(xToken, process.env.TOKEN_SECRET);
+    req.user = decoded;
+    next();
+  } catch (err) {
+    throw new ClientError(401, 'authentication required');
+  }
```

Comment:
You can skip this try catch block, because express will automatically catch the jwt error and handle it with the error middleware.